# HG changeset patch
# User Rik <rik@octave.org>
# Date 1452983560 28800
#      Sat Jan 16 14:32:40 2016 -0800
# Node ID 397f6ee4a35d008ac7a057214c38e0838d39cf09
# Parent  529593df22faccc04c6631b37a10fcc45fe77e5f
Extend msg from gripe_disabled_feature to handle case where libraries not available.

* gripes.h (gripe_disabled_feature): Use 'fcn' rather than 'func' for
variable name.

* gripes.cc (gripe_disabled_feature): Use 'fcn' rather than 'func' for
variable name.  Change message to "support for %s was unavailable or disabled
when %s was built".

diff --git a/libinterp/corefcn/gripes.cc b/libinterp/corefcn/gripes.cc
--- a/libinterp/corefcn/gripes.cc
+++ b/libinterp/corefcn/gripes.cc
@@ -50,21 +50,21 @@ void
 gripe_data_file_in_path (const std::string& fcn, const std::string& file)
 {
   warning_with_id ("Octave:data-file-in-path",
                    "%s: '%s' found by searching load path",
                    fcn.c_str (), file.c_str ());
 }
 
 void
-gripe_disabled_feature (const std::string& func, const std::string& feature,
+gripe_disabled_feature (const std::string& fcn, const std::string& feature,
                         const std::string& pkg /*="Octave"*/)
 {
-  error ("%s: support for %s was disabled when %s was built",
-         func.c_str (), feature.c_str (), pkg.c_str ());
+  error ("%s: support for %s was unavailable or disabled when %s was built",
+         fcn.c_str (), feature.c_str (), pkg.c_str ());
 }
 
 void
 gripe_divide_by_zero (void)
 {
   warning_with_id ("Octave:divide-by-zero", "division by zero");
 }
 
diff --git a/libinterp/corefcn/gripes.h b/libinterp/corefcn/gripes.h
--- a/libinterp/corefcn/gripes.h
+++ b/libinterp/corefcn/gripes.h
@@ -39,17 +39,17 @@ void gripe_2_or_3_dim_plot (void);
 
 OCTAVE_NORETURN OCTINTERP_API extern
 void gripe_data_conversion (const char *from, const char *to);
 
 extern OCTINTERP_API void
 gripe_data_file_in_path (const std::string& fcn, const std::string& file);
 
 OCTAVE_NORETURN OCTINTERP_API extern
-void gripe_disabled_feature (const std::string& func,
+void gripe_disabled_feature (const std::string& fcn,
                              const std::string& feature,
                              const std::string& pkg="Octave");
 
 extern OCTINTERP_API void
 gripe_divide_by_zero (void);
 
 extern OCTINTERP_API void
 gripe_empty_arg (const char *name, bool is_error);
